hack/verify-staticcheck.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+kube::golang::verify_go_version
+
+FOCUS="${1:-}"
+
+# TODO: Start verifying more checks!
+# See https://staticcheck.io/docs/checks
+CHECKS=(
+  "U1*" # Detect unused code
+)
+export IFS=','; CHECKS="${CHECKS[*]}"; unset IFS
```

Comment:
What is this for?  Are you sure it is right?  `"${VAR[*]}"` is a smell - should be `[@]` but I can't tell what you're trying to do..

Reply:
Are you trying to convert CHECKS from an array to a string?  You're leaving it declared as an array:

```
thockin@freakshow2 kubernetes master /$ CHECKS=( "U1*" "FOO" "BAR" )
thockin@freakshow2 kubernetes master /$ export IFS=','; CHECKS="${CHECKS[*]}"; unset IFS
thockin@freakshow2 kubernetes master /$ for x in "${CHECKS[@]}"; do echo $x; done
U1*,FOO,BAR
FOO
BAR
thockin@freakshow2 kubernetes master /$ declare -p CHECKS
declare -a CHECKS=([0]="U1*,FOO,BAR" [1]="FOO" [2]="BAR")
```

Reply:
Sorry, accidentally pushed an intermediate version, fixed now.

CHECKS is declared as an array so the individual elements can be commented on, but the actual command line needs a comma-delimited string

Reply:
[SA1019](https://staticcheck.io/docs/checks#SA1019) "Using a deprecated function, variable, constant or field" is annoying, but maybe still worth including?